code/modules/mob/living/basic/minebots/minebot_abilities.dm

```diff
@@ -49,3 +49,118 @@
+		return FALSE
+	owner.Shake(duration = wind_up_timer)
+	addtimer(CALLBACK(src, PROC_REF(launch_missile), target_turf), wind_up_timer)
+	StartCooldown()
+	return TRUE
+
+/datum/action/cooldown/mob_cooldown/missile_launcher/proc/launch_missile(turf/target_turf)
+	new /obj/effect/temp_visual/mook_dust(get_turf(owner))
+	var/obj/effect/temp_visual/rising_rocket/new_rocket = new(get_turf(owner))
+	addtimer(CALLBACK(src, PROC_REF(drop_missile), target_turf), new_rocket.duration)
+
+/datum/action/cooldown/mob_cooldown/missile_launcher/proc/drop_missile(turf/target_turf)
+	new /obj/effect/temp_visual/falling_rocket(target_turf)
+	var/obj/effect/temp_visual/falling_shadow = new /obj/effect/temp_visual/shadow_telegraph(target_turf)
+	animate(falling_shadow, transform = matrix().Scale(0.1, 0.1), time = falling_shadow.duration)
+
+/datum/action/cooldown/mob_cooldown/drop_landmine
+	name = "Landmine"
+	button_icon = 'icons/obj/weapons/grenade.dmi'
+	button_icon_state = "landmine"
+	background_icon_state = "bg_default"
+	overlay_icon_state = "bg_default_border"
+	desc = "Drop a landmine!"
+	cooldown_time = 10 SECONDS
+	shared_cooldown = NONE
+	melee_cooldown_time = 0 SECONDS
+	click_to_activate = FALSE
+
+/datum/action/cooldown/mob_cooldown/drop_landmine/Activate(atom/target)
+	var/turf/my_turf = get_turf(owner)
+	if(isopenspaceturf(my_turf))
+		return FALSE
+	var/obj/effect/mine/minebot/my_mine = new(my_turf)
+	my_mine.ignore_list = owner.faction.Copy()
+	playsound(my_turf, 'sound/weapons/armbomb.ogg', 20)
+	StartCooldown()
+	return TRUE
+
+/obj/effect/temp_visual/rising_rocket
+	name = "Missile"
+	icon = 'icons/obj/weapons/guns/projectiles.dmi'
+	icon_state = "84mm-heap"
+	layer = ABOVE_ALL_MOB_LAYER
+	duration = 2 SECONDS
+
+/obj/effect/temp_visual/rising_rocket/Initialize(mapload)
+	. = ..()
+	playsound(src.drop_location(), 'sound/weapons/minebot_rocket.ogg', 100, FALSE)
+	animate(src, pixel_y = base_pixel_y + 500, time = duration, easing = EASE_IN)
+
+/obj/effect/temp_visual/falling_rocket
+	name = "Missile"
+	icon = 'icons/obj/weapons/guns/projectiles.dmi'
+	icon_state = "84mm-heap"
+	layer = ABOVE_ALL_MOB_LAYER
+	duration = 0.7 SECONDS
+	pixel_y = 60
+	///the radius of our explosion
+	var/explosion_radius = 2
+	///damage of our explosion
+	var/explosion_damage = 100
+
+/obj/effect/temp_visual/falling_rocket/Initialize(mapload)
+	. = ..()
+	transform = transform.Turn(180)
+	addtimer(CALLBACK(src, PROC_REF(create_explosion)), duration)
+	animate(src, pixel_y = 0, time = duration)
+
+/obj/effect/temp_visual/falling_rocket/proc/create_explosion()
+	playsound(src.drop_location(), 'sound/weapons/minebot_rocket.ogg', 100, FALSE)
+	var/datum/effect_system/fluid_spread/smoke/smoke = new
+	smoke.set_up(1, holder = src)
+	smoke.start()
+	for(var/turf/target_turf as anything in RANGE_TURFS(explosion_radius, src))
+		var/mob/living/living_target = locate() in target_turf
+		if(isnull(living_target))
+			continue
+		living_target.apply_damage(explosion_damage)
```

Comment:
It seems weird that this can only hit one mob per turf, is that intentional? 

Also I think this can technically hit incorporeal mobs like Revenants